### PR TITLE
feat: match multiple original sites in url cache

### LINKED ISSUE #315: add support for url cache entries with multiple original sites
Port of https://github.com/adobe/helix-sidekick-extension/pull/835

### PR DESCRIPTION
**Pull request by rofe**
Fix #315

### CODE REVIEW

File: src/extension/project.js [hunk 1]
```diff
@@ -210,7 +210,7 @@ export function assembleProject({
   } else {
     giturl = `https://github.com/${owner}/${repo}/tree/${ref}`;
   }
-  const id = `${owner}/${repo}/${ref}`;
+  const id = `${owner}/${repo}`;
```
Comment (rofe) [inline src/extension/project.js:213]: This change is not directly related to this issue, but we currently only ever add projects with the `main` ref, so this suffx in the ID is obsolete.

File: test/mocks/discover.js [hunk 2]
```diff
@@ -10,41 +10,73 @@
  * governing permissions and limitations under the License.
  */
 
-import sinon from 'sinon';
+// @ts-ignore
+import fetchMock from 'fetch-mock/esm/client.js';
```
Comment (rofe) [inline test/mocks/discover.js:14]: Refactored to use `fetchMock`

**Review by dylandepass — APPROVED** [review]
